Makefile

```diff
@@ -174,6 +174,10 @@ export SPARK_PYTHON_IMAGE=docker.io/apache/spark-py:v$(SPARK_PYTHON_VERSION)
 GO_LICENSES_VERSION=v1.6.0
 GO_LICENSES_BIN=$(TOOLS_DIR)/go-licenses
 
+# ginkgo
+GINKGO_VERSION=v2.17.1
```

Comment:
the version of `ginkgo` is defined by `go.mod`, and it would be better to keep them consistency. Hence, we don't need to define the specific version in makefile.

Reply:
Great. Delete the version of ginkgo in makefile.

Reply:
Will `go install` honor the go.mod file?

Reply:
It's better to match what's in go.mod instead. Perhaps extract from go.mod using a short bit of shell and explicitly install that version instead.

Reply:
This should extract it:

`GINKGO_VERSION := $(shell grep ginkgo go.mod | head -1 | awk '{print $$2}')`

Reply:
@craigcondit according to the docs (https://go.dev/ref/mod#go-install)
```
# Install a program at the version selected by the module in the current directory.
$ go install golang.org/x/tools/gopls
```
and

```
If the arguments don’t have version suffixes, go install may run in module-aware mode or GOPATH mode, depending on the GO111MODULE environment variable and the presence of a go.mod file.
```

Also, I have test `go install github.com/onsi/ginkgo/v2/ginkgo` on my local. the version is equal to what we define in go.mod

Reply:
Ok, I stand corrected.